### DISCUSSION TITLE
Order of stimuli appears to alternate for no reason (multi-likert scale plugin)

### DISCUSSION THREAD

**Discussion by nickriches** [jspsych/jsPsych#2770] [Q&A] [answered]
Hi. I am using the multi-likert scale plugin. A sentence is presented, and then two rating scales. Code is below (NB the variables `sentence` `scale1` and `scale2` are read in from a javascript file).

When I run this, the sentence should appear, then the `scale1` stimulus, then the first likert scale, then the `scale2` stimulus, then the second likert scale. However, the `sentence` stimulus alternates appearing first before `scale1` then before `scale2`, then back to its original position and so on. 

(NB I tried to create a Minimal Worked Example, and ran into a few problems. I hope the code below is sufficient).

Thanks

```javascript

var rating_scale_1_to_20 = [
    "Not at all", "1", "3", "4", "5", "6", "7", "8", "9", "10", "11", "12", "13", "14", "15", "16", "17", "18", "19", "Completely"
  ]

var ratings = {
  type: jsPsychSurveyLikert,
  questions: [
    {prompt: () => String(jsPsych.timelineVariable(`sentence`)) +
              `<br>` +
              String(jsPsych.timelineVariable(`scale1`)),
              labels: rating_scale_1_to_20,
              required: true},
    {prompt: () => String(jsPsych.timelineVariable(`scale2`)),
              labels: rating_scale_1_to_20,
              required: true}
  ],
  scale_width: 800,
  data: {
    likert_response: 'ratings_response'
  },
  randomize_question_order: true
};


var ratings_test_procedure = {
  timeline: [ratings],
  timeline_variables: comparative_stims,
  randomize_order: true
};



timeline.push(ratings_test_procedure);




    /* start the experiment */
    jsPsych.run(timeline);
```

**Comment by ChristopheBossens** [comment, marked as answer]
Could not test the code because I do not have the `comparative_stims `variable. However, could the ordering that that you see be related to the fact that the `randomize_question_order `property in your likert scale is set to true?

**Reply by nickriches** [reply]
Thanks for the responses and sorry I missed it. Github normally shows a blue dot at the top right when there is a response to a query you made on a github forum, but something happened.

I've just managed to solve it, and it doesn't appear to be related to the question order. The error rates to the string function. When I pulled the second stimulus inside the first string function, behaviour is as expected. I am not a sufficient expert in js to know how / why this works, but it does!

```js
var ratings = {
  type: jsPsychSurveyLikert,
  questions: [
    {prompt: () => String(jsPsych.timelineVariable(`sentence`)) +
              `<br>` + jsPsych.timelineVariable(`scale1`),
              labels: rating_scale_1_to_20,
              required: true},
    {prompt: () => String(jsPsych.timelineVariable(`scale2`)),
              labels: rating_scale_1_to_20,
              required: true}
  ],
  scale_width: 800,
  data: {
    likert_response: 'ratings_response'
  },
  randomize_question_order: true
};

```

**Reply by nickriches** [reply]
Okay. Facepalm. You were absolutely right. Changing randomize_question_order to false fixed it. I thought that this variable related to the order of stimuli rather than the order of the questions, which is why I set it to true. Of course the randomizing of the order of stimuli is handled elsewhere. Thanks for your help.
